lib/vanagon/project.rb

```diff
@@ -692,5 +717,32 @@ def load_upstream_settings(upstream_project_name, upstream_git_url, upstream_git
         upstream_project.cleanup
       end
     end
+
+    # Load the settings hash for the current project/platform combination from a
+    # yaml file as produced by `publish_yaml_settings`. file:// and http:// URIs
+    # are accepted. If the URI uses http://, a sha1 URI is also required.
+    #
+    # @param yaml_settings_uri [String] A URI to a yaml settings file
+    # @param yaml_settings_sha1_uri [String] A URI to a sha1sum file for the yaml settings file
+    # @raise [Vanagon::Error] when the settings file can't be found
+    def load_yaml_settings(yaml_settings_uri, yaml_settings_sha1_uri = nil)
+      source_type = Vanagon::Component::Source.determine_source_type(yaml_settings_uri)
+      if yaml_settings_uri =~ /^file/ && source_type == :unknown
+        raise Vanagon::Error, "Tried to load YAML settings from '#{yaml_settings_uri}', but the file doesn't exist"
+      end
+
+      unless %i[http local].include?(source_type)
+        raise Vanagon::Error, "YAML settings can only be inherited from a file or http source (got #{yaml_settings_uri})"
+      end
+
+      Dir.mktmpdir do |working_directory|
+        source = Vanagon::Component::Source.source(yaml_settings_uri,
+                                                   workdir: working_directory,
+                                                   sum: yaml_settings_sha1_uri,
+                                                   sum_type: 'sha1')
+        source.fetch
+        @settings.merge!(YAML.load(File.read(File.join(working_directory, source.file)))) # rubocop:disable Security/YAMLLoad
```

Comment:
Is there a reason we need to use `YAML.load` instead of `YAML.safe_load`?

Reply:
I had thought that Psych flat-out disallowed symbol keys in when using `safe_load`, but it looks like you can explicitly allow them -- I'll do that!